### DISCUSSION TITLE
Tesla Solar/Powerwall Data Logger

### DISCUSSION THREAD

**Discussion by natrlhy** [teslamate-org/teslamate#877] [Ideas]
I know it's completely unrelated, but is there potential to add tesla solar/powerwall to teslamate or is there another active project for this? I don't have Tesla Solar or a Powerwall yet, but I'm very likely going to get both before the end of the year. I'd love to have a single point with all this data, like... TeslaMate! 🥇

**Comment by adriankumpf** [comment]
It is practically impossible for me to support Tesla solar/powerwall, as I do not have one. Maybe someone knows some projects for this though?

**Comment by jrester** [comment]
I am the maintainer of the powerwall python API which is also used in Homeassistant. I think it would be quite easy to rewrite the API in elixir and integrate it into Teslamate.

**Comment by jrester** [comment]
I published a [first version](https://hex.pm/packages/tesla_powerwall/0.1.0) of the API. It only includes some basic features and I need to expand the documentation and testing.
@adriankumpf if you want to I can try to integrate it into teslamate.

**Comment by adriankumpf** [comment]
Cool! I'm open for suggestions and how such an integration would look like.

**Comment by jameshiscott** [comment]
super interested in this one, would be nice to see that the charge is coming from the powerwall or from the grid (so the price can be zero or the current price)

**Comment by jrester** [comment]
I am wondering whether it would be better to directly implement this into TeslaMate or develop an extra service that broadcasts the data over MQTT and have TeslaMate consume the data and work with it. I am quite unsure about this.
What do you think @adriankumpf?

**Comment by natrlhy** [comment]
I’d love to see it fully integrated with TeslaMate. The name and platform for self hosted users is perfect. If we could somehow create a separate docker container that would show some cool web status page similar to what the Tesla App shows, that’d be awesome. I still don’t have Powerwalls/solar installed yet so tough for me to test, but that’s my vision to have a fully web based UI that one can see/visualize their solar/powerwall data.

Historical exports/graphs, etc as well ;)

**Comment by denisjoshua** [comment]
I would like a lot to have it also.
:-) 
Thanks

**Comment by natrlhy** [comment]
> I am wondering whether it would be better to directly implement this into TeslaMate or develop an extra service that broadcasts the data over MQTT and have TeslaMate consume the data and work with it. I am quite unsure about this.
> What do you think @adriankumpf?

We now have solar and Powerwalls installed! Anything I can do to help develop this @jrester

**Comment by blharveymi** [comment]
Any update on this? anything can do to help?

**Reply by s0schen** [reply]
I am interested in this as well.   I'm loving Teslamate and would like this integrated as I am getting powerwalls soon.

However I did find two other projects:
https://github.com/mihailescu2m/powerwall_monitor
https://github.com/liveaverage/docker-powerwall-dashboard

**Comment by superforno** [comment]
I'm here to help testing it

**Comment by epheterson** [comment]
Also would like to see this, and happy to help! Mostly interested in logging battery capacity over time to track loss.

**Comment by matteoandreozzi** [comment]
@adriankumpf what happened to this thread? I am up for testing the PowerWall integration (and contribute to the code too) if that's still of interest.
